parser/src/string.rs

```diff
@@ -21,7 +22,7 @@ use rustpython_parser_core::{
 const MAX_UNICODE_NAME: usize = 88;
 
 struct StringParser<'a> {
-    chars: std::iter::Peekable<std::str::Chars<'a>>,
+    chars: MultiPeek<std::str::Chars<'a>>,
```

Comment:
I would prefer if we don't have to use `MultiPeek`. We can instead clone the `chars` iterator. Cloning is cheap because it only requires coping the start end end slice pointers (and I'm somewhat certain that LLVM can figure out that it only needs to copy the start pointer). 

Using `clone` would allow us to remove the `Peekable` altogether, similar to what we do in the Lexer's Cursor:

https://github.com/astral-sh/RustPython-Parser/blob/7cbec096e6032f370d42022593021f5cd1b65021/parser/src/lexer/cursor.rs#L30-L42